File: libnetwork/src/main/java/tim/com/libnetwork/network/okhttp/call/HCall.java
package tim.com.libnetwork.network.okhttp.call;

import androidx.lifecycle.LifecycleOwner;
import java.util.concurrent.Executor;
import retrofit2.Call;

public interface HCall<T> extends Call<T> {
    // 绑定生命周期
    HCall<T> bindLifecycle(LifecycleOwner lifecycleOwner);
    // 设置 callbackExecutor
    HCall<T> callbackExecutor(Executor mExecutor);
}

File: libnetwork/src/main/java/tim/com/libnetwork/network/okhttp/call/HCall.kt
package tim.com.libnetwork.network.okhttp.call;

import androidx.lifecycle.LifecycleOwner
import retrofit2.Call
import java.util.concurrent.Executor

interface HCall<T> :Call<T> {
    //绑定生命周期
    fun bindLifecycle(lifecycleOwner: LifecycleOwner) :HCall<T>
    //设置callbackExecutor
    fun callbackExecutor(mExecutor: Executor):HCall<T>

}